packages/primitives/src/components/Combobox/Combobox.tsx

```diff
@@ -0,0 +1,764 @@
+      onClick={composeEventHandlers(iconProps.onClick, () => {
+        // Whilst browsers generally have no issue focusing the trigger when clicking
+        // on a label, Safari seems to struggle with the fact that there's no `onClick`.
+        // We force `focus` in this case. Note: this doesn't create any other side-effect
+        // because we are preventing default in `onPointerDown` so effectively
+        // this only runs for a label "click"
+        context.trigger?.focus();
+      })}
+      onPointerDown={composeEventHandlers(iconProps.onPointerDown, (event) => {
+        // only call handler if it's the left button (mousedown gets triggered by all mouse buttons)
+        // but not when the control key is pressed (avoiding MacOS right click)
+        if (event.button === 0 && event.ctrlKey === false) {
+          handleOpen();
+          // prevent trigger from stealing focus from the active item after opening.
+          event.preventDefault();
+        }
+      })}
+      onKeyDown={composeEventHandlers(iconProps.onKeyDown, (event) => {
+        if (OPEN_KEYS.includes(event.key)) {
+          handleOpen();
+          event.preventDefault();
+        }
+      })}
+    >
+      {children || '▼'}
+    </button>
+  );
+});
+
+/* -------------------------------------------------------------------------------------------------
+ * ComboboxPortal
+ * -----------------------------------------------------------------------------------------------*/
+
+const PORTAL_NAME = 'ComboboxPortal';
+
+type IPortalProps = React.ComponentPropsWithoutRef<typeof PortalPrimitive>;
+interface PortalProps extends Omit<IPortalProps, 'asChild'> {
+  children?: React.ReactNode;
+}
+
+const ComboboxPortal: React.FC<PortalProps> = (props) => {
+  return <PortalPrimitive asChild {...props} />;
+};
+
+ComboboxPortal.displayName = PORTAL_NAME;
+
+/* -------------------------------------------------------------------------------------------------
+ * ComboboxContent
+ * -----------------------------------------------------------------------------------------------*/
+
+const CONTENT_NAME = 'ComboboxContent';
+
+type ComboboxContentElement = ComboboxContentImplElement;
+type ContentProps = ComboboxContentImplProps;
+
+const ComboboxContent = React.forwardRef<ComboboxContentElement, ContentProps>((props, forwardedRef) => {
+  const context = useComboboxContext(CONTENT_NAME);
+  const [fragment, setFragment] = React.useState<DocumentFragment>();
+
+  // setting the fragment in `useLayoutEffect` as `DocumentFragment` doesn't exist on the server
+  React.useLayoutEffect(() => {
+    setFragment(new DocumentFragment());
+  }, []);
+
+  if (!context.open) {
+    const frag = fragment as Element | undefined;
+    return frag
+      ? ReactDOM.createPortal(
+          <Collection.Slot scope={undefined}>
+            <div>{props.children}</div>
+          </Collection.Slot>,
+          frag,
+        )
+      : null;
```

Comment:
No I don't think we should because we can't add a `fragment` as the mounting point. It would also mean the DS relies on Primitives but Primitives relies on the DS. Its also not used when the menu is open.

Reply:
True. Wondering if we should move `Portal` to `primitives` too at some point?

Reply:
We could do, we'd have to overload the API to allow passing a container as a prop & if one is not supplied, set one up. I'd do this as a separate ticket i think :)